src/runtime/test_case_runner.js

```diff
@@ -194,7 +216,19 @@ export default class TestCaseRunner {
     } else if (this.isSkippingSteps()) {
       return { status: Status.SKIPPED }
     }
-    return this.invokeStep(step, stepDefinitions[0])
+    return this.runStepHooks(step, this.beforeStepHookDefinitions).then(
+      responses => {
+        const errors = responses.filter(response => response.exception)
+        if (errors && errors.length) {
+          return {
+            exception: 'BeforeStep hook failed', // TODO return stacktrace(s)?
```

Comment:
I'm curious how we should handle multiple exceptions? And should I instead check if status is `failed` here instead of checking for `exception` in any of the responses? Or both?